src/Scanner/TimeNodeScanner.ts

```diff
@@ -43,22 +43,22 @@ export default class TimeNodeScanner extends ChainScanner implements ITimeNodeSc
     return this.scanning;
   }
 
-  public stop(): boolean {
+  public async stop(): Promise<boolean> {
     if (this.scanning) {
       this.scanning = false;
       // Clear scanning intervals.
       clearInterval(this.cacheInterval);
       clearInterval(this.chainInterval);
 
-      this.txPool.stop();
+      await this.txPool.stop();
 
       // Mark that we've stopped.
       this.config.logger.info('Scanner STOPPED');
     }
 
     Object.keys(this.buckets).forEach((bucketType: string) => {
-      Object.keys(this.buckets[bucketType]).forEach((key: string) => {
-        this.stopWatcher(this.buckets[bucketType][key]);
+      Object.keys(this.buckets[bucketType]).forEach(async (key: string) => {
```

Comment:
I think you need to switch to `for ... of` loop because `forEach` won't wait, when iterator function is async. So, basically after you finish iterating over array you won't be sure if all watchers were stopped if using `map` or `forEach`.